### PR TITLE
Closed Loops

### PR DESCRIPTION
**Pull request by jacbn**
Enables dragging of all movable points in closed loops without buggy behaviour

### CODE REVIEW

File: src/GraphUtils.ts [hunk 1]
```diff
@@ -330,14 +333,18 @@ export function findInterceptX(canvasProperties: CanvasProperties, pts: Point[])
...
     const intercepts = [];
 
-    if (pts[0].y == canvasProperties.centerPx.y) intercepts.push(pts[0]);
-    for (let i = 1; i < pts.length; i++) {
-        if (pts[i].y == canvasProperties.centerPx.y) {
-            intercepts.push(new Point(pts[i].x, pts[i].y));
+    for (let i = 0; i < pts.length-1; i++) {
+        if (pts[i].y === canvasProperties.centerPx.y) {
+            const start = i;
+            for (; i < pts.length && pts[i].y === canvasProperties.centerPx.y; i++);
```
Comment (skyepurchase) [inline src/GraphUtils.ts]: Incrementing `i` within an anonymous for loop body would be more readable, but this approach may be more efficient. Up to you.

File: src/GraphUtils.ts [hunk 2]
```diff
@@ -356,8 +363,13 @@ export function findInterceptY(canvasProperties: CanvasProperties, pts: Point[])
 
     if (pts[0].x == canvasProperties.centerPx.x) intercepts.push(pts[0]);
     for (let i = 1; i < pts.length; i++) {
-        if (pts[i].x == canvasProperties.centerPx.x) {
-            intercepts.push(new Point(pts[i].x, pts[i].y));
+        if (pts[i].x === canvasProperties.centerPx.x) {
+            const start = i;
+            for (; i < pts.length && pts[i].x === canvasProperties.centerPx.x; i++);
```
Comment (skyepurchase) [inline src/GraphUtils.ts]: Same as above, incrementing within the body would be more readable

**Review by skyepurchase — APPROVED** [review]
LGTM

File: src/GraphUtils.ts (hunk 1 above)
Comment (jacbn) [reply src/GraphUtils.ts]: I imagine it will compile the same, so we'll go for readability

**Pull request merged**